modules/memberships/class-jetpack-memberships.php

```diff
@@ -392,7 +431,8 @@ public function register_gutenberg_block() {
 			Blocks::jetpack_register_block(
 				'jetpack/recurring-payments',
 				array(
-					'render_callback' => array( $this, 'render_button' ),
+					'render_callback' =>  array( $this, 'render_button' ),
+					'uses_context' => array( 'isPremiumContentChild' ),
```

Comment:
The goal here is to make `$block->context` available in the render callback. This is possible when using `register_block_from_metadata` to register the block (see Social Link block as an example).

I stepped through the implementation of `register_block_type_from_metadata`, and it has the effect of passing this `uses_context` setting through to `register_block_type`. I'm attempting here to hard-code this for Recurring Payments, since I've had difficulty restructuring this block to use a `block.json` like Social Link and getting the correct path for that file once it is created.